node/src/lib.rs

```diff
@@ -312,7 +331,7 @@ pub(crate) type NewFull = subspace_service::NewFull<
 #[derive(Derivative)]
 #[derivative(Debug)]
 #[must_use = "Node should be closed"]
-pub struct Node {
+pub struct Node<F: Farmer> {
```

Comment:
Can you elaborate on why `Node` requires a `Farmer`?

Reply:
Because we need to know at compile time for what kind of farmer we wanna build `Node` here. In future I feel that we want to rework `Farmer` trait, but right now it is kind of stub for decoupling node and farmer.